src/pages/index.js

```diff
@@ -51,37 +63,136 @@ const profileEditModal = new PopupWithForm(
   handleProfileFormSubmit
 );
 
+const editAvatarModal = new PopupWithForm(
+  "#change-profile-picture-modal",
+  handleAvatarFormSubmit //create a function of this name that runs after I submit the form//
+);
+
+const avatarUpdateValidator = new FormValidator(config, updateAvatarForm);
+
+const deleteConfirmationModal = new PopupWithConfirmation(
+  "#delete-confirmation-modal"
+);
+
 const addCardModal = new PopupWithForm("#add-card-modal", handleCardAddSubmit);
 
 export const previewImageModal = new PopupWithImage("#previewImageModal");
+
 // call setEventListeners
 
 const profileUserInfo = new UserInfo(
   ".profile__title",
-  ".profile__description"
-);
-
-const cardSection = new Section(
-  { items: initialCards, renderer: createCard },
-  ".cards__list"
+  ".profile__description",
+  ".profile__image"
 );
 
 //functions
 
 function handleProfileFormSubmit(formData) {
-  console.log(formData);
-  profileUserInfo.setUserInfo(formData.title, formData.description);
+  //fetch request to change the name and description on the server
+  profileEditModal.setButtonText("Saving...");
+  //profileEditModal.reset();
+  api
+
+    .updateUserInfo(formData)
+    .then((res) => {
+      profileUserInfo.setUserInfo(res.name, res.about);
+      profileEditModal.close();
+      //modal__form.reset();
+    })
+
+    .catch((err) => {
+      console.error(err);
+      alert("Failed to change user info");
+      console.log(formData);
+    })
+    .finally(() => {
+      profileEditModal.setButtonText("Save");
+    });
 }
 
 function handleCardAddSubmit(inputValues) {
   const cardData = { name: inputValues.title, link: inputValues.url };
-  cardSection.addItem(cardData);
-  addFormValidator.resetValidation();
+  addCardModal.setButtonText("Saving...");
+  api
+    .addCard(cardData)
+    .then((res) => {
+      const cardElement = createCard(res);
+      cardSection.addItem(cardElement);
+      addFormValidator.resetValidation();
+    })
+    .finally(() => {
+      addCardModal.setButtonText("Save");
+    });
+  addCardModal.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.